share/functions/__fish_print_interfaces.fish

```diff
@@ -1,8 +1,6 @@
 function __fish_print_interfaces --description "Print a list of known network interfaces"
     if test -d /sys/class/net
-        for i in /sys/class/net/*
-            basename $i
-        end
+        string replace /sys/class/net/ '' /sys/class/net/*
```

Comment:
I've just realized that if `/sys/class/net` is empty, fish will display an error. A workaround would be to use `set`.

```fish
set -l interfaces /sys/class/net/*
string replace /sys/class/net/ '' $interfaces
```